source/distributed/fully_distributed_tria.cc

```diff
@@ -0,0 +1,525 @@
+
+      dealii::Triangulation<dim, spacedim>::create_triangulation(vertices,
+                                                                 cells,
+                                                                 subcelldata);
+    }
+
+
+
+    template <int dim, int spacedim>
+    Triangulation<dim, spacedim>::Triangulation(MPI_Comm       mpi_communicator,
+                                                const Settings settings)
+      : parallel::DistributedTriangulationBase<dim, spacedim>(
+          mpi_communicator,
+          (settings & construct_multigrid_hierarchy) ?
+            static_cast<
+              typename dealii::Triangulation<dim, spacedim>::MeshSmoothing>(
+              dealii::Triangulation<dim>::none |
+              Triangulation<dim,
+                            spacedim>::limit_level_difference_at_vertices) :
+            static_cast<
+              typename dealii::Triangulation<dim, spacedim>::MeshSmoothing>(
+              dealii::Triangulation<dim>::none),
+          false)
+      , settings(settings)
+      , create_triangulation_for_internal_usage(false)
+      , prepare_coarsening_and_refinement_for_internal_usage(false)
+    {}
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::update_number_cache()
+    {
+      parallel::Triangulation<dim, spacedim>::update_number_cache();
+
+      if (settings & construct_multigrid_hierarchy)
+        parallel::Triangulation<dim, spacedim>::fill_level_ghost_owners();
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::execute_coarsening_and_refinement()
+    {
+      AssertThrow(false, ExcNotImplemented());
+    }
+
+
+
+    template <int dim, int spacedim>
+    bool
+    Triangulation<dim, spacedim>::prepare_coarsening_and_refinement()
+    {
+      AssertThrow(prepare_coarsening_and_refinement_for_internal_usage,
+                  ExcMessage("No coarsening and refinement is supported!"));
+
+      return dealii::Triangulation<dim, spacedim>::
+        prepare_coarsening_and_refinement();
+    }
+
+
+
+    template <int dim, int spacedim>
+    bool
+    Triangulation<dim, spacedim>::has_hanging_nodes() const
+    {
+      AssertThrow(false, ExcNotImplemented());
+      return false;
+    }
+
+
+
+    template <int dim, int spacedim>
+    std::size_t
+    Triangulation<dim, spacedim>::memory_consumption() const
+    {
+      std::size_t mem =
+        this->dealii::parallel::TriangulationBase<dim, spacedim>::
+          memory_consumption() +
+        MemoryConsumption::memory_consumption(
+          coarse_cell_id_to_coarse_cell_index_vector) +
+        MemoryConsumption::memory_consumption(
+          coarse_cell_index_to_coarse_cell_id_vector);
+      return mem;
+    }
+
+
+
+    template <int dim, int spacedim>
+    std::map<unsigned int, std::set<dealii::types::subdomain_id>>
+    Triangulation<dim, spacedim>::compute_vertices_with_ghost_neighbors() const
```

Comment:
Inside of `parallel::TriangulationBase` we can find an implementation and a **comment** regarding the correctness of this function:
```cpp
   template <int dim, int spacedim>
   std::map<unsigned int, std::set<::types::subdomain_id>>
   TriangulationBase<dim, spacedim>::compute_vertices_with_ghost_neighbors()
     const
   {
     // TODO: we are not treating periodic neighbors correctly here. If we do
     // we can remove the overriding implementation for p::d::Triangulation
     // that is currently using a p4est callback to get correct ghost neighbors
     // over periodic faces.
```
I would not mind if we/I fix this function (by taking some ideas from my implementation) and then remove the implementation from `p:f:t`.

However, it would be nice if someone could help me and point out what the issue with this function is and if it was already solved by my function. Which test could help me, if this function works properly or not? I guess either @tjhei has implemented this function or knows the right person...

I'll clean up my code so that it will be clear what it does (in addition).

ping @bangerth @kronbichler

Reply:
I see. Let's leave this for a follow-up then. Can you open a github issue that describes the problem?

Reply:
I have added comments and have restructured the code. See also the issue #8734.